packages/web-components/fast-foundation/src/listbox/listbox.ts

```diff
@@ -160,18 +160,38 @@ export class Listbox extends FASTElement {
     protected focusAndScrollOptionIntoView(): void {
         if (this.contains(document.activeElement) && this.firstSelectedOption) {
             this.firstSelectedOption.focus();
-            this.firstSelectedOption.scrollIntoView({ block: "nearest" });
+            requestAnimationFrame(() => {
+                this.firstSelectedOption.scrollIntoView({ block: "nearest" });
+            });
         }
     }
 
+    /**
+     * @internal
+     */
+    private shouldSkipFocus: boolean = false;
+
     /**
      * @internal
      */
     public focusinHandler(e: FocusEvent): void {
-        if (e.target === e.currentTarget) {
+        if (!this.shouldSkipFocus && e.target === e.currentTarget) {
             this.setSelectedOptions();
             this.focusAndScrollOptionIntoView();
         }
+
+        this.shouldSkipFocus = false;
+    }
+
+    /**
+     * Prevents `focusin` events from firing before `click` events when the
+     * element is unfocused.
+     *
+     * @internal
+     */
+    public mousedownHandler(e: MouseEvent): boolean | void {
+        this.shouldSkipFocus = !this.contains(document.activeElement);
```

Comment:
Would the reference to document.activeElement possibly be incorrect if this its in the shadow root?